src/cc/bcc_syms.cc

```diff
@@ -158,6 +158,15 @@ int ProcSyms::_add_module(const char *modname, uint64_t start, uint64_t end,
     auto module = Module(
         modname, check_mount_ns ? ps->mount_ns_instance_.get() : nullptr,
         &ps->symbol_option_);
+
+    // pid/maps doesn't account for file_offset of text within the ELF.
+    // It only gives the mmap offset. We need the real offset for symbol
+    // lookup.
+    if (module.type_ == ModuleType::SO) {
+      module.elf_so_offset_ = bcc_elf_get_text_file_offset(modname);
+      module.elf_so_addr_ = bcc_elf_get_text_file_addr(modname);
```

Comment:
The `.text` section name is hardcoded in the function. For SO with customer linker, this may or may not be true. I guess there is no need for error handling here since we are targeting the default settings.